pkg/backend/filestate/meta_test.go

```diff
@@ -36,21 +35,51 @@ func TestEnsurePulumiMeta(t *testing.T) {
 			// Empty bucket should be initialized to
 			// the current version by default.
 			desc: "empty",
+			want: pulumiMeta{Version: 1},
+		},
+		{
+			// Use legacy mode even for the new bucket
+			// by setting an environment variable.
+			desc: "empty/legacy",
+			env:  map[string]string{PulumiFilestateLegacyLayoutEnvVar: "1"},
 			want: pulumiMeta{Version: 0},
 		},
 		{
-			desc: "version 0",
+			desc: "empty/legacy/true",
+			env:  map[string]string{PulumiFilestateLegacyLayoutEnvVar: "true"},
+			want: pulumiMeta{Version: 0},
+		},
+		{
+			desc: "empty/legacy/false",
+			env:  map[string]string{PulumiFilestateLegacyLayoutEnvVar: "false"},
+			want: pulumiMeta{Version: 1},
+		},
```

Comment:
Nit: Would appreciate a small comment explaining that you expect it to upgrade the pulumiMeta version.